services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/server/grpc/GoalStateProvisionerServer.java

```diff
@@ -144,9 +171,24 @@ public void blockUntilShutdown() throws InterruptedException {
             return new StreamObserver<Goalstate.GoalStateV2>() {
                 @Override
                 public void onNext(Goalstate.GoalStateV2 value) {
+                    Span pSpan = tracer.activeSpan();
+                    Span span;
+
+                    if(pSpan != null){
+                        span = tracer.buildSpan("alcor-ncm-server-pushdown-gs").asChildOf(pSpan.context()).start();
+                    }else{
+                        span = tracer.buildSpan("alcor-ncm-server-pushdown-gs").start();
+                    }
+                    logger.log(Level.INFO, "[pushGoalStatesStream] Got parent span: "+(null == pSpan? "null" : pSpan.toString()));
+                    logger.log(Level.INFO, "[pushGoalStatesStream] Built child span: "+span.toString());
+
+                    Scope cscope = tracer.scopeManager().activate(span);
 
                     logger.log(Level.INFO, "pushGoalStatesStream : receiving GS V2 message " + value.toString());
                     long start = System.currentTimeMillis();
+                    Span storeGsSpan = tracer.buildSpan("alcor-ncm-server-store-gs").asChildOf(span.context()).start();
```

Comment:
same here. The span name should be in a project or service config file, and do a reference here.